### PR TITLE
Update HuaweiCloud provider default region and allocate EIP for head node

### PR DESCRIPTION
**Pull request by kiwik**
1. "ap-southeast-3" region is in Singapore, we can use more stable
and speed networking access to some resources.
2. Update server flavor and image to match the region.
3. Allocate and attach EIP to head node
4. Add workspace security group egress rule
5. Add workspace subnet DNS option
6. Add configurable workspace bandwidth option for EIP and NAT

Related-with: #1011

### CODE REVIEW

File: python/cloudtik/providers/_private/huaweicloud/utils.py [hunk 1]
```diff
@@ -234,6 +238,10 @@ def get_default_huaweicloud_cloud_storage(provider_config):
     return cloud_storage_info
 
 
+def get_huaweicloud_workspace_bandwidth_conf():
```
Comment (jerrychenhf) [inline python/cloudtik/providers/_private/huaweicloud/utils.py:241]: Suggest to use workspace config file in provider section to allow user to specify any provider specific options like this.

For example,
provider:
    # bandwidth for public ip in MB
    public_ip_bandwidth: 20

Comment (kiwik) [reply python/cloudtik/providers/_private/huaweicloud/utils.py:241]: I think we need two public ip bandwidth options for workspace NAT and node, apply them when creating workspace NAT and creating head node, I will add them in next commit. Thank you.

Comment (jerrychenhf) [reply python/cloudtik/providers/_private/huaweicloud/utils.py]: Agree. Two options look good.

**Comment by kiwik** [comment]
rebase on main branch

**Pull request merged**